libraries/lts/include/mcrl2/lts/detail/liblts_bisim_m.h

```diff
@@ -0,0 +1,579 @@
+     *           Based on the preprint "Computing minimal distinguishing Hennessey-Milner formulas is NP-hard
+     *           . But variants are tractable", 2023 by Jan Martens and Jan Friso Groote
+     *  \param[in] s The state number for which the resulting formula should be true
+     *  \param[in] t The state number for which the resulting formula should be false
+     *  \return A minimal observation depth distinguishing state formula, that is often also minimum negation-depth and irreducible. */
+    mcrl2::state_formulas::state_formula dist_formula_mindepth(const std::size_t s, const std::size_t t) {
+        formula f = distinguish(block_index_of_a_state[s], block_index_of_a_state[t]);
+        mCRL2log(mcrl2::log::info) << "done with formula \n";
+        return convertFormula(f);
+    };
+
+    bool in_same_class(const std::size_t s, const std::size_t t) 
+    {
+        return block_index_of_a_state[s] == block_index_of_a_state[t]; 
+    }
+
+
+private:
+    typedef std::size_t block_index_type;
+    typedef std::size_t state_type;
+    typedef std::size_t level_type;
+    typedef std::size_t formula_index_type;
+    typedef std::size_t label_type;
+    state_type initial_l2;
+
+    state_type max_state_index;
+    LTS_TYPE& aut;
+    std::set <block_index_type> partition;
+
+    struct block
+    {
+        state_type state_index;                   // The state number that represent the states in this block
+        block_index_type block_index;             // The sequence number of this block.
+        block_index_type parent_block_index;      // Index of the parent block.
+        level_type level;
+
+
+        // If there is no parent block, this refers to the block itself.
+        std::vector < state_type > states;
+        std::vector < transition > transitions;
+        std::set<std::pair < label_type, block_index_type >> outgoing_observations;
+
+        void swap(block& b)
+        {
+            state_type state_index1 = b.state_index;
+            b.state_index = state_index;
+            state_index = state_index1;
+
+            block_index_type block_index1 = b.block_index;
+            b.block_index = block_index;
+            block_index = block_index1;
+
+            block_index_type parent_block_index1 = b.parent_block_index;
+            b.parent_block_index = parent_block_index;
+            parent_block_index = parent_block_index1;
+
+            level_type level1 = b.level;
+            b.level = level;
+            level = level1;
+
+            states.swap(b.states);
+            transitions.swap(b.transitions);
+        }
+    };
+
+    struct formula 
+    {
+        formula_index_type index;
+        label_type label;
+        bool negated;
+        std::vector <formula> conjunctions;
+        std::set <block_index_type > truths;
+
+        int depth() {
+            int max_depth = 0;
+            for (formula f : conjunctions) {
```

Comment:
Generally curly braces after functions and for loops, and multi-line if statements should start on a new line.